### PR TITLE
Pds api 153: error messaging

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Now returns error messages as result when things go awry.

## ⚙️ Test Data and/or Report
```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_in::1.0' 
{"request":"/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_in::1.0","message":"The lidvid urn:nasa:pds:izenberg_pdart14_meap:document:ns_in was not found"}


$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_in::1.0'  --header 'Accept:text/xml'
<error><request>/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_in::1.0</request><message>The lidvid urn:nasa:pds:izenberg_pdart14_meap:document:ns_in was not found</message></error>
```

## ♻️ Related Issues
NASA-PDS/registry-api#443

### CODE REVIEW

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Ready for review -- will fix any sonatype errors if they appear

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	@Override
+	protected ErrorMessage readInternal(Class<? extends ErrorMessage> clazz, HttpInputMessage inputMessage)
+			throws IOException, HttpMessageNotReadableException { return new ErrorMessage(); }
+
+	@Override
+	protected void writeInternal(ErrorMessage t, HttpOutputMessage outputMessage)
+			throws IOException, HttpMessageNotWritableException
+	{
+		OutputStreamWriter writer = new OutputStreamWriter(outputMessage.getBody());
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/HtmlErrorMessageSerializer.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	protected ErrorMessage readInternal(Class<? extends ErrorMessage> clazz, HttpInputMessage inputMessage)
+			throws IOException, HttpMessageNotReadableException { return new ErrorMessage(); }
+
+	@Override
+	protected void writeInternal(ErrorMessage t, HttpOutputMessage outputMessage)
+			throws IOException, HttpMessageNotWritableException
+	{
+		OutputStreamWriter osw = new OutputStreamWriter(outputMessage.getBody());
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/HtmlErrorMessageSerializer.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/JsonErrorMessageSerializer.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	@Override
+	protected ErrorMessage readInternal(Class<? extends ErrorMessage> clazz, HttpInputMessage inputMessage)
+			throws IOException, HttpMessageNotReadableException { return new ErrorMessage(); }
+
+	@Override
+	protected void writeInternal(ErrorMessage t, HttpOutputMessage outputMessage)
+			throws IOException, HttpMessageNotWritableException
+	{
+        OutputStreamWriter wr = new OutputStreamWriter(outputMessage.getBody());
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/JsonErrorMessageSerializer.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/XmlErrorMessageSerializer.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+	@Override
+	protected ErrorMessage readInternal(Class<? extends ErrorMessage> clazz, HttpInputMessage inputMessage)
+			throws IOException, HttpMessageNotReadableException { return new ErrorMessage(); }
+
+	@Override
+	protected void writeInternal(ErrorMessage t, HttpOutputMessage outputMessage)
+			throws IOException, HttpMessageNotWritableException
+	{
+		OutputStreamWriter osw = new OutputStreamWriter(outputMessage.getBody());
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/XmlErrorMessageSerializer.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		OutputStreamWriter writer = new OutputStreamWriter(outputMessage.getBody(), "UTF-8");
+		writer.write ("request,message\n");
+		writer.write('"');
+		writer.write(t.getRequest());
+		writer.write("\",\"");
+		writer.write(t.getMessage());
+		writer.write('"');
+		writer.write('\n');
+		writer.close();
+	}
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java:43]: help

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

I do not understand the resource leak error/message. It is closed and on the stack so why is it a resource leak when the function ends? Can you point me at what is expected? Google always takes me to the glossy ads and the console does not tell me anything new or have detailed explanations.

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java:43]: @sonatype-lift help

**Comment by al-niessner** [comment]
> @tloubrieu-jpl
> 
> I do not understand the resource leak error/message. It is closed and on the stack so why is it a resource leak when the function ends? Can you point me at what is expected? Google always takes me to the glossy ads and the console does not tell me anything new or have detailed explanations.

Nevermind. I think I figured out what they are complaining about.

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/XmlErrorMessageSerializer.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+			throws IOException, HttpMessageNotWritableException
+	{
+		OutputStreamWriter osw = new OutputStreamWriter(outputMessage.getBody(), "UTF-8");
+		osw.write("<error><request>");
+		osw.write(t.getRequest());
+		osw.write("</request><message>");
+		osw.write(t.getMessage());
+		osw.write("</message></error>");
+		osw.close();
+	}
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/XmlErrorMessageSerializer.java:41]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/JsonErrorMessageSerializer.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+			throws IOException, HttpMessageNotWritableException
+	{
+        OutputStreamWriter wr = new OutputStreamWriter(outputMessage.getBody(), "UTF-8");
+        wr.write("{\"request\":\"");
+        wr.write(t.getRequest());
+        wr.write("\",\"message\":\"");
+        wr.write(t.getMessage());
+        wr.write("\"}");
+        wr.close();
+	}
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/JsonErrorMessageSerializer.java:43]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/HtmlErrorMessageSerializer.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+			throws IOException, HttpMessageNotWritableException
+	{
+		OutputStreamWriter osw = new OutputStreamWriter(outputMessage.getBody(), "UTF-8");
+		osw.write("<html><body><h1>Error Message</h1><h2>From Request</h2><p>");
+		osw.write(t.getRequest());
+		osw.write("</p><h2>Message</h2><p>");
+		osw.write(t.getMessage());
+		osw.write("</p></body></html>");
+		osw.close();
+	}
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/HtmlErrorMessageSerializer.java:40]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/CsvErrorMessageSerializer.java:43]: fixed

**Review by tloubrieu-jpl — APPROVED** [review]
Perfect, thanks @al-niessner

**Pull request merged**